Userland/Services/SpiceAgent/Message.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2023, Caoimhe Byrne <caoimhebyrne06@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "Message.h"
+#include <AK/MemoryStream.h>
+#include <AK/Stream.h>
+#include <AK/String.h>
+
+namespace SpiceAgent {
+
+ErrorOr<void> Message::write_to_stream(AK::Stream&)
+{
+    return Error::from_string_literal("write_to_stream not implemented!");
+}
+
+ErrorOr<String> Message::debug_description()
+{
+    return TRY("Message {}"_string);
+}
+
+ErrorOr<AnnounceCapabilitiesMessage> AnnounceCapabilitiesMessage::read_from_stream(AK::Stream& stream)
+{
+    // If this message is a capabilities request, we don't have to parse anything else
+    auto is_requesting = TRY(stream.read_value<u32>()) == 1;
+    if (is_requesting) {
+        return AnnounceCapabilitiesMessage(is_requesting);
+    }
+
+    return Error::from_string_literal("Unexpected non-requesting announce capabilities message received!");
+}
+
+ErrorOr<void> AnnounceCapabilitiesMessage::write_to_stream(AK::Stream& stream)
+{
+    TRY(stream.write_value<u32>(is_requesting()));
+    for (auto capability : capabilities()) {
+        TRY(stream.write_value(1 << to_underlying(capability)));
+    }
```

Comment:
I don't see the issue with it... what do you mean a "variable-length" bitfield? I'm pretty sure someone said the opposite to me about it before...

Reply:
Assuming that we have the capabilities 0, 1, 2, and 3, if we want to announce capabilities 0, 1, and 3, I'm pretty sure what we should be writing is 0x0000000b, since it has the zeroeth, first and third bit set. What you are currently writing (as far as I can tell) is 0x00000001, 0x00000002, 0x00000008.